### PR TITLE
Button & Spinner Component for React Native

### PR DESCRIPTION
**Pull request by l2succes**
This PR ports the Button and Spinner component from emission into palette. This also sets some guidelines on how to share common code and interfaces between React Native and React Web version of a component.

Note this works best for components that behave essentially the same on both platforms and can thus share a common interface.

Kapture 2019-06-15 at 1 00 25

Design spec:

Screen Shot 2019-06-15 at 1 09 51 AM

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit e042eea4155f0b1dae34bfb96e9b42968eb67160

https://deploy-preview-515--artsy-palette.netlify.com

**Comment by damassi** [comment]
Looks perfect 👌

**Comment by damassi** [comment]
Ahh, looks like this was pretty far out of date with master. Lot's of failures after merge. Looking...

**Comment by artsyit** [comment]
:rocket: PR was released in v4.14.17 :rocket:

File: packages/palette/src/elements/Spinner/Spinner.ios.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  startRotation() {
+    this.rotation.setValue(0)
+    Animated.timing(this.rotation, {
+      toValue: 1,
+      duration: 1000,
+      easing: Easing.linear,
+      useNativeDriver: true,
+    }).start(() => this.startRotation())
+  }
```
Comment (alloy) [inline packages/palette/src/elements/Spinner/Spinner.ios.tsx]: Nice 👌

**Pull request merged**